test/martha/martha_v3.test.js

```diff
@@ -279,30 +536,86 @@ test.serial('martha_v3 calls bond Bond with the "fence" provider when the Data O
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'drs://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), jadeDosMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Gen3 CRDC response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(gen3CrdcResponse);
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'drs://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
     const result = response.send.lastCall.args[0];
     t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
-    t.deepEqual(Object.assign({}, result), fullExpectedResult(null));
+    t.deepEqual(Object.assign({}, result), gen3CrdcDrsMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses BDC response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(bdcDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), bdcDrsMarthaResult);
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Anvil response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(anvilDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), anvilDrsMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Kids First response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(kidsFirstDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), kidsFirstDrsMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses HCA response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(hcaDosResponse);
```

Comment:
Per my suggestion below, I looked how to do this in Sinon, if you are interested in pursuing it. Per https://sinonjs.org/releases/v9.0.2/stubs/ :
```suggestion
    getJsonFromApiStub.withArgs([fill in args here]).onFirstCall().resolves(hcaDosResponse);
```
You just need to fill in what the params should be to `getJsonFrom` when called from the url `dos://jade.datarepo-dev.broadinstitute.org/abc`.